### ISSUE TITLE
GetForecastAsync does not handle API responses properly

### ISSUE THREAD

**Issue by dsoronda** [amweiss/dark-sky-core#10]
Code only handles proper responses otherwise throws JSON parse exception.

Using invalid key, DarkSky API returns "Invalid" and that response can't be deserialized into Forecast response.

Using uri `forecast/716c78f54adfbe480b8168907f743729/42,9150,- 78,7410?&exclude=flags&extend=hourly&lang=x-pig-latin&units=si` response is "Not found" and it's not handled properly.

**Comment by amweiss** [comment]
What would you expect to happen? The forecast is invalid and throwing the exception indicates that.

**Comment by dsoronda** [comment]
Return error status like IsSuccessful = false with error description from DarkSky, not directly try to serialize invalid response.

I'm currently working on fork, and this is simple workaround that I used

~~~csharp
            string responseContent = await response.Content.ReadAsStringAsync();

            var darkSkyResponse = new DarkSkyResponse()
            {
                IsSuccessful = response.IsSuccessStatusCode
                // TODO : add error message back to response
            };

            if (darkSkyResponse.IsSuccessful)
            {
                long callsParsed;

                darkSkyResponse.Response = JsonConvert.DeserializeObject<Forecast>(responseContent);
...
            }
~~~

**Comment by dsoronda** [comment]
I'm working on this on my fork.

**Comment by amweiss** [comment]
Merged in #11
